sdk/core/azure-core/src/main/java/com/azure/core/implementation/serializer/HttpResponseBodyDecoder.java

```diff
@@ -274,90 +234,46 @@ private static Object deserializePage(String value, Type resultType, Type wireTy
      */
     private static Object convertToResultType(Object wireResponse, Type resultType, Type wireType) {
         Object result = wireResponse;
-
-        if (wireResponse != null) {
-            if (resultType == byte[].class) {
-                if (wireType == Base64Url.class) {
-                    result = ((Base64Url) wireResponse).decodedBytes();
-                }
-            } else if (resultType == OffsetDateTime.class) {
-                if (wireType == DateTimeRfc1123.class) {
-                    result = ((DateTimeRfc1123) wireResponse).getDateTime();
```

Comment:
Not sure why we get the and return the `OffsetDateTime` from `DateTimeRfc1123` and `UnixTime`